features/include/pcl/features/normal_3d.h

```diff
@@ -414,9 +401,16 @@ namespace pcl
       /** whether the sensor origin of the input cloud or a user given viewpoint should be used.*/
       bool use_sensor_origin_;
 
+      /** \brief The number of threads the scheduler should use. */
+      unsigned int threads_;
+
     public:
       PCL_MAKE_ALIGNED_OPERATOR_NEW
   };
+
+  template <typename PointInT, typename PointOutT>
+  using NormalEstimationOMP PCL_DEPRECATED(1, 12, "use NormalEstimation instead") = NormalEstimation<PointInT, PointOutT>;
```

Comment:
@SergioRAgostinho Would you prefer this approach or the approach where the `setNumberOfThreads` is visible only in the derived class made for OMP?

Reply:
If the point is to try to enforce consistency, the latter. Only OMP classes need to worry about how many threads are running.

Reply:
@shrijitsingh99 Could you please `undeprecate` the OMP class?

Reply:
I didn't understand the discussion. What do you mean by derived class for OMP?

To clarify, these PRs are meant as basework for integration with executors. The ideas was to get the design of this combined class approved then integrate with executors then merge with master.
Apologize for that, I should have clarified it early on.

Reply:
> then integrate with executors then merge with master.

If I understood Sergio correctly, OMP class should not be deprecated till the executors are integrated.

> What do you mean by derived class for OMP?

The OMP class derives from the base class. It adds only a ctor and overloads where needed to expose controls to OMP which are hidden in the base class.

Benefits:
* doesn't break API to users till executors are ready
* removes 2 implementations and any chance of them diverging
* still has 1 class for you to add executors to

Reply:
Oh okay so that's just removing the deprecation warning.

I got confused with something else then.

Reply:
* Remove deprecation warnings
* Let OMP class handle the num_thread related stuff (setting/changing, etc.)